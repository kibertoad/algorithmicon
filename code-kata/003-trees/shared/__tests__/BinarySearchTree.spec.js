const { BinarySearchTree } = require('../BinarySearchTree')
const { BinarySearchTreeNode } = require('../BinarySearchTreeNode.js')

describe('BinarySearchTree', () => {
  it('correctly adds new element', () => {
    const tree = new BinarySearchTree()
    tree.addValue(70)
    tree.addValue(75)
    tree.addValue(40)
    tree.addValue(20)
    tree.addValue(60)
    tree.addValue(80)
    tree.addValue(90)

    expect(tree.toString()).toMatchSnapshot()
  })

  it('in-order traversal', () => {})
})
