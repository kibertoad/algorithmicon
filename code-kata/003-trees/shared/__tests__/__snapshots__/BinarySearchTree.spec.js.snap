// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BinarySearchTree correctly adds new element 1`] = `
"Node 70. Left: 40. Right: 75
Node 40. Left: 20. Right: 60
Node 20. Left: null. Right: null
Node 60. Left: null. Right: null
Node 75. Left: null. Right: 80
Node 80. Left: null. Right: 90
Node 90. Left: null. Right: null
"
`;
